pyoutline/outline/plugins/local.py

```diff
@@ -33,66 +35,92 @@
 USE_LOCAL_CORES = 1
 USE_LOCAL_THREADS = 0
 
+
 def init_cuerun_plugin(cuerun):
-    cuerun.get_parser().add_plugin_option("-L", "--run-local",
-                                          action="callback",
-                                          type="int",
-                                          nargs=1,
-                                          help="Run local cores",
-                                          callback=setup_event,
-                                          callback_args=(cuerun,))
-
-    cuerun.get_parser().add_plugin_option("-T", "--run-local-threads",
-                                          action="callback",
-                                          type="int",
-                                          nargs=1,
-                                          help="Set number of threads for local cores to run per frame.",
-                                          callback=setup_local_threads,
-                                          callback_args=(cuerun,))
+    """Initialize the cuerun plugin."""
+
+    cuerun.get_parser().add_plugin_option(
+        "-L", "--run-local",
+        action="callback",
+        type="int",
+        nargs=1,
+        help="Run local cores",
+        callback=setup_event,
+        callback_args=(cuerun,))
+
+    cuerun.get_parser().add_plugin_option(
+        "-T", "--run-local-threads",
+        action="callback",
+        type="int",
+        nargs=1,
+        help="Set number of threads for local cores to run per frame.",
+        callback=setup_local_threads,
+        callback_args=(cuerun,))
 
 
 def setup_event(option, opt, value, parser, *args, **kwargs):
+    """Main callback for the plugin, to set up the event listener."""
+    del option
+    del opt
+    del parser
+    del kwargs
+
     # Don't have a way to pass this to the event but
     # a global will do.
+    # pylint: disable=global-statement
     global USE_LOCAL_CORES
     USE_LOCAL_CORES = value
 
     args[0].add_event_listener(event.BEFORE_LAUNCH, setup_local_cores)
 
 
 def setup_local_threads(option, opt, value, parser, *args, **kwargs):
+    """Callback for the --run-local-threads option."""
+
+    del option
```

Comment:
Why are the arguments being deleted here?

Reply:
The function arguments are standard -- passed by the `OptionParser` callback system -- so we don't want to change them, but the function itself doesn't need these arguments.

These arguments get flagged by pylint as unused arguments that should be removed, and the deletion resolves those issues. I'm not sure how standard that is exactly, but I've seen it done in past projects.

I think it's also considered good practice anyway for unused arguments, in that it makes it explicit to future developers that those arguments aren't being used.